### PR TITLE
[fix](function)return NULL rather than 'null' if path not found

### LINKED ISSUE #25102: [Bug] function JSON_EXTRACT return 'null' rather than NULL if cannot find result by extract_path
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

1.2 & trunk

### What's Wrong?

image

### What You Expected?

return NULL for such case

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Yulei-Yang**
## Proposed changes

Issue Number: close #25102

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run p0

File: be/src/vec/functions/function_json.cpp [hunk 1]
```diff
@@ -880,6 +885,37 @@ class FunctionJson : public IFunction {
...
+class FunctionJsonNullable : public IFunction {
+public:
+    static constexpr auto name = Impl::name;
+    static FunctionPtr create() { return std::make_shared<FunctionJsonNullable<Impl>>(); }
+    String get_name() const override { return name; }
+    size_t get_number_of_arguments() const override { return 0; }
+    bool is_variadic() const override { return true; }
+    bool use_default_implementation_for_constants() const override { return true; }
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return std::make_shared<DataTypeString>();
```
Comment (xy720) [inline be/src/vec/functions/function_json.cpp:898]:
```suggestion
        return make_nullable(std::make_shared<DataTypeString>());
```

**Comment by xy720** [comment]
please try this case:
SELECT json_extract('[1, 2, 3]', '$.[1]');

File: be/src/vec/functions/function_json.cpp (hunk 1 above)
Comment (Yulei-Yang) [reply be/src/vec/functions/function_json.cpp]: fixed

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
> please try this case: SELECT json_extract('[1, 2, 3]', '$.[1]');

fixed

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run clickbench

**Comment by Yulei-Yang** [comment]
run clickbench

**Comment by xy720** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run external

**Comment by Yulei-Yang** [comment]
run external

**Comment by Yulei-Yang** [comment]
run external

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run external

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run pipelinex_p0

**Review by xy720 — APPROVED** [review]
LGTM

**Review by yuxuan-luo — APPROVED** [review]

**Comment by xiaokang** [comment]
@Yulei-Yang can you submit a pr for branch-2.0

**Pull request merged**